### ISSUE TITLE
Labs (experimental features)

### ISSUE THREAD

**Issue by DrewAPicture** [AffiliateWP/AffiliateWP#2060]
PR: #2059

The idea of Labs is simple: Make it possible to test-drive experimental, yet production-ready features without immediately committing to maintaining them as add-ons or core features.

**Benefits**

The implementation would take the form of pairing a free add-on with core, which gives us a few big upsides:

* Labs can be iterated independent of core, which means it won't have to be dependent on a particular core release
* Unlike a beta, Labs features would be production-ready, meaning anybody who installs the add-on can run, test, and give feedback on them
* We can solicit feedback and make enabling features contingent on usage data collection.
* The Labs plugin can be run by any AffWP user

**Core Functionality**
The Labs feature registration API would be built into core, while Labs itself would handle registering features. The attached PR covers implementation in core only. Development of the free add-on would happen in the [affiliatewp-labs](https://github.com/AffiliateWP/affiliatewp-labs) repo.

In a nutshell, Labs could be our testing ground where we could collect real, actionaable feedback and determine customer buy-in before it either gets spun out as a free or pro add-on, or even merged into core in a future release.

---

If we decide to do this, I would propose we roll it into core in 2.0.4 so it will already be in place for the 2.1 release.

**Comment by DrewAPicture** [comment]
Notably, the PR also abstracts out the logic for the Batch Processor Registry into a new base `Registry` class that's reusable (it's reused for registering features).

**Comment by DrewAPicture** [comment]
Punting this to 2.2 (but not Future Release). After some thought I think the implementation needs further consideration.
